src/peep/peep.c

```diff
@@ -3421,7 +3464,532 @@ static void peep_update_fixing(int steps, rct_peep* peep){
+		return;
+	}
+
+	peep_update_action(); // TODO: arguments
+}
+
+static void loc_6C133F(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	invalidate_sprite_2((rct_sprite *) peep);
+
+	int out = peep_update_action(0, 0, 0, peep);// TODO: arguments
+	if (out != 0) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	sint16 dx = peep->z;
+	sprite_move(unknown, unkown, dx, peep); // TODO: arguments
+
+	invalidate_sprite_2((rct_sprite *) peep);
+}
+
+static void loc_6C13B4(int steps, rct_peep *peep, rct_ride *ride) {
+	if (peep->action == 0xFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	peep_update_action(0, 0, 0, peep);// TODO: arguments
+}
+
+static void loc_6C13F8(int steps, rct_peep *peep, rct_ride *ride) {
+	if (peep->action == 0xFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	peep_update_action(0, 0, 0, peep);// TODO: arguments
+	if (peep->action_frame == 0x28) {
+		ride->mechanic_status = 4;
+		ride->window_invalidate_flags &= 0x20;
+	}
+
+	if (peep->action_frame == 0x13 ||
+		peep->action_frame == 0x19 ||
+		peep->action_frame == 0x1F ||
+		peep->action_frame == 0x25 ||
+		peep->action_frame == 0x2B) {
+
+		audio_play_sound_panned(SOUND_MECHANIC_FIX, AUDIO_PLAY_AT_LOCATION, peep->x, peep->y, peep->z);
+	}
+}
+
+static void loc_6C14E5(int steps, rct_peep *peep, rct_ride *ride) {
+	invalidate_sprite_2((rct_sprite *) peep);
+	int out = peep_update_action(0, 0, 0, peep);// TODO: arguments
+	if (out == 0) {
+		// jnb
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	sprite_move(unknown, unkown, dx, peep); // TODO: arguments
+	invalidate_sprite_2((rct_sprite *) peep);
+}
+
+static void loc_6C1559(int steps, rct_peep *peep, rct_ride *ride) {
+	if (peep->action != 0xFF) {
+		peep_update_action(0, 0, 0, peep); // TODO: arguments
+		return;
+	}
+
+	// loc_6C156F:
+	ride_fix_breakdown(peep->current_ride, 0); // TODO: What should be the reliabilityIncreaseFactor?
+
+	loc_6C1653(steps, peep, ride);
+}
+
+static void loc_6C15EF(int steps, rct_peep *peep, rct_ride *ride, uint32 ebp) {
+	invalidate_sprite_2(peep);
+	int out = peep_update_action(0, 0, 0, peep); // TODO: arguments
+	if (out == 0) {
+		// jnb
+		// loc_6C1644:
+		return;
+	}
+
+	uint32 ebx = peep->current_ride_station;
+	uint16 dx = ride->station_heights[ebx];
+	dx = dx << 3;
+
+	ebp = 0;
+	if (ebp == 0x10) {
+		sprite_move(); // TODO: arguments
+		invalidate_sprite_2((rct_sprite *) peep);
+		return;
+	}
+
+	// TODO: Figure out what happens at 0x6C1626
+	rct_sprite *sprite = &g_sprite_list[ride->type];
+}
+
+void loc_6C1653(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	int edx = peep->sub_state;
+	uint32 ebp = RCT2_GLOBAL(0x992a38, uint32);
+
+	if (peep->state != 0x17) {
+		ebp = ride->breakdown_reason_pending;
+		ebp = RCT2_ADDRESS(0x992A18, uint32)[ride->breakdown_reason_pending];
+	}
+
+	// loc_6C167B
+	do {
+		edx++;
+	} while ((ebp >> edx) & 0b1 != 1);
```

Comment:
Although what you have is technically correct the rest of the code base tends to be in the following format.
`(ebp & (1<< edx)) == 0`